### PR TITLE
gh-101100: Improve docs on exception attributes

### PR DESCRIPTION
**Pull request by AlexWaygood**
* Issue: gh-101100

----
📚 Documentation preview 📚: https://cpython-previews--113057.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/exceptions.rst [hunk 1]
```diff
@@ -791,7 +794,8 @@ Exception Objects
    it.  There is no type check to make sure that *cause* is either an exception
    instance or ``None``.  This steals a reference to *cause*.
 
-   :attr:`__suppress_context__` is implicitly set to ``True`` by this function.
+   :attr:~BaseException.`__suppress_context__` is implicitly set to ``True`` by
```
Comment (AlexWaygood) [inline Doc/c-api/exceptions.rst]:
```suggestion
   :attr:`~BaseException.__suppress_context__` is implicitly set to ``True`` by
```

File: Doc/library/exceptions.rst [hunk 2]
```diff
@@ -38,36 +38,45 @@ information on defining exceptions is available in the Python Tutorial under
...
+
+   This implicit exception context can be
+   supplemented with an explicit cause by using :keyword:`!from` with
+   :keyword:`raise`::
+
+      raise new_exc from original_exc
+
+   The expression following :keyword:`from<raise>` must be an exception or ``None``. It
+   will be set as :attr:`!__cause__` on the raised exception. Setting
+   :attr:!`__cause__` also implicitly sets the :attr:`!__suppress_context__`
```
Comment (AlexWaygood) [inline Doc/library/exceptions.rst]:
```suggestion
   :attr:`!__cause__` also implicitly sets the :attr:`!__suppress_context__`
```

File: Doc/library/traceback.rst [hunk 3]
```diff
@@ -255,27 +257,31 @@ capture data for later printing in a lightweight fashion.
...
 
-      A :class:`TracebackException` of the original ``__cause__``.
+      A :class:`!TracebackException` of the original
+      :attr:`~BaseException.__cause__`.
 
    .. attribute:: __context__
 
-      A :class:`TracebackException` of the original ``__context__``.
+      A :class:`!TracebackException` of the original
+      :attr`~BaseException.__context__`.
```
Comment (AlexWaygood) [inline Doc/library/traceback.rst]:
```suggestion
      :attr:`~BaseException.__context__`.
```

File: Doc/reference/simple_stmts.rst [hunk 4]
```diff
@@ -592,11 +592,13 @@ same exception instance, with its traceback set to its argument), like so::
 The ``from`` clause is used for exception chaining: if given, the second
 *expression* must be another exception class or instance. If the second
 expression is an exception instance, it will be attached to the raised
-exception as the :attr:`__cause__` attribute (which is writable). If the
+exception as the :attr:`~BaseException.__cause__` attribute (which is writable). If the
 expression is an exception class, the class will be instantiated and the
 resulting exception instance will be attached to the raised exception as the
-:attr:`__cause__` attribute. If the raised exception is not handled, both
-exceptions will be printed::
+:attr:!`__cause__` attribute. If the raised exception is not handled, both
```
Comment (AlexWaygood) [inline Doc/reference/simple_stmts.rst]:
```suggestion
:attr:`!__cause__` attribute. If the raised exception is not handled, both
```

**Review by willingc — APPROVED** [review]
Thanks @AlexWaygood. Nice improvements 👍🏼

**Pull request merged**